test/metabase/query_processor_test/alternative_date_test.clj

```diff
@@ -225,15 +228,33 @@
                  (mt/rows (mt/dataset string-times
                             (qp/process-query
                               (assoc (mt/mbql-query times)
-                                     :middleware {:format-rows? false})))))))))
+                                     :middleware {:format-rows? false}))))))))
+
+      (testing "mongo only supports datetime"
+        (mt/test-drivers #{:mongo}
+          (is (= [[(t/instant "2004-10-19T10:23:54Z")]
+                  [(t/instant "2008-10-19T10:23:54Z")]
+                  [(t/instant "2012-10-19T10:23:54Z")]]
+               (mt/rows (mt/dataset string-times
+                          (qp/process-query
+                           (assoc (mt/mbql-query times
+                                                 {:fields [$ts]})
+                                  :middleware {:format-rows? false})))))))))
+
     (testing "are queryable as dates"
       (testing "a datetime field"
         ;; TODO: why does this fail on oracle? gives a NPE
         (mt/test-drivers (disj (sql-jdbc.tu/sql-jdbc-drivers) :oracle :sparksql)
           (is (= 1
                  (count (mt/rows (mt/dataset string-times
                                    (mt/run-mbql-query times
-                                     {:filter   [:= [:datetime-field $ts :day] "2008-10-19"]}))))))))
+                                     {:filter   [:= [:datetime-field $ts :day] "2008-10-19"]})))))))
+        (mt/test-drivers #{:mongo}
+          (is (= 1
+                 (count (mt/rows (mt/dataset string-times
+                                   (mt/run-mbql-query times
+                                     {:filter   [:= [:datetime-field $ts :day] "2008-10-19"]
```

Comment:
`:datetime-field` is legacy MBQL syntax, you shouldn't write any new tests with it. If you touch a test with it, it's probably a good idea to update it 

```suggestion
                                     {:filter   [:= [:field %ts {:temporal-unit :day}] "2008-10-19"]
```

Actually you could use the MBQL shorthand and do

```suggestion
                                     {:filter   [:= !day.ts "2008-10-19"]
```

which doesn't win points for obviousness but it sure is easy to write.